UI-8299 - Support notebooks in the Content Migration Tool - update README (#99)
activeviam/atoti-ui-migration: README.md

# Atoti UI Migration

A command line interface to migrate:

> - an Atoti UI application content that was saved in older versions of Atoti UI to be compatible with newer versions.

Reviewer: In particular, I'd like to avoid _"an"_ at the beginning of the sentence, to make it clearer that this is a generic tool, not dedicated to a specific Atoti project.

```suggestion
- content saved in an old version of Atoti UI, making it compatible with newer versions.
```

> - an Atoti jupyter notebook that was saved in older versions of Atoti.

Reviewer: ```suggestion
- Atoti Jupyter notebooks saved in an old version of Atoti, making them compatible with newer versions.
```

cd migration
```

The migration CLI requires two files:

> - the file to migrate: it can be the server content stored in a JSON file, _content.json_ or an Atoti jupyter notebook, _notebook.ipynb_.

Reviewer: ```suggestion
- the file to migrate: it can be the server content stored in a JSON file, _content.json_ or an Atoti Jupyter notebook, _notebook.ipynb_.
```

- a JSON file with the Atoti servers' information: _servers.json_

Let's create these files.

> ## Migrate a server content

Reviewer: ```suggestion
## Migrate a content server 
```

Finally open your dashboards, widgets, filters, ... Make sure they work as expected.

> ## Migrate an Atoti jupyter notebook

Reviewer: ```suggestion
## Migrate an Atoti Jupyter notebook
```

> To migrate a jupyter notebook, two files are necessary:

Reviewer: ```suggestion
To migrate a Jupyter notebook, two files are necessary:
```

> - _notebook.ipynb_, the notebook to migrate. It is really important that the notebook has the `.ipynb` extension.

Reviewer: I'd like to avoid _"really important"_, because I find it too vague. I find _"must"_ enough. Alternatively, we could say something like _"ipynb" is the only supported file extension.

```suggestion
- _notebook.ipynb_, the notebook to migrate. This file must have the `.ipynb` extension.
```

- _servers.json_, the JSON file containing the server information.

To get the servers information, follow the [download the servers information section](#download-the-servers-information) instructions.

> Now that your migration folder contains all the required file, you can run the following CLI:

Reviewer: _"Command"_ is more correct than CLI, because CLI is for _Command line interface_. It's an interface that can contain multiple commands (like a REST interface can contain multiple endpoints).

```suggestion
Now that your migration folder contains all the required files, you can run the following command:
```

```
npx -- activeui-migration migrate -i notebook.ipynb -o migrated-notebook.ipynb -s servers.json --from-version 0.7 --to-version 0.8
```

This command generates a file named `migrated-notebook.json` in the same folder. It contains the migrated notebook with all its migrated widgets, ready to be used with the Atoti version you migrated to. You can see the [run the CLI section](#run-the-cli) to get more details about the command line and its options.

> :warning: Calculated measures are not handled by the migration. In that case, you should update manually the widgets containing the calculated measures.

Reviewer: I'd like to be a bit more specific and avoid generic words like _"handle"_ or _"update"_, in order to make the sentence clearer. In this case, we are talking about removing saved calculated measures from widgets queries.

```suggestion
:warning: Saved calculated measures are not removed from the queries of the widgets in your notebook. If you migrate your Content Server on top of your notebook, then you have to remove these saved calculated measures from the queries in your notebook manually (if any).
```
